changelogs/current.yaml

```diff
@@ -113,6 +113,13 @@ new_features:
     Added support for dropping stats via
     :ref:`DropAction <envoy_v3_api_msg_extensions.stat_sinks.open_telemetry.v3.SinkConfig.DropAction>` during
     custom metric conversion.
+- area: upstream
+  change: |
+    Added support for setting ``TCP_NOTSENT_LOWAT`` socket option on upstream connections via
+    :ref:`tcp_notsent_lowat <envoy_v3_api_field_config.cluster.v3.UpstreamConnectionOptions.tcp_notsent_lowat>`
+    in cluster configuration or
+    :ref:`tcp_notsent_lowat <envoy_v3_api_field_extensions.upstreams.tcp.v3.TcpProtocolOptions.tcp_notsent_lowat>`
```

Comment:
I'm not sure why we need two places to set upstream setting. Can we also add this to downstream as well? It can be also limit buffer bloat on the response stream.